bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/handler/IpCameraGroupHandler.java

```diff
@@ -0,0 +1,414 @@
+                        }
+                        cameraOrder.add(handler);
+                    }
+                }
+            }
+        }
+    }
+
+    // Event based. This is called as each camera comes online after the group handler is registered.
+    public void cameraOnline(String uid) {
+        logger.debug("New camera {} came online, checking if part of this group", uid);
+        if (config.get(CONFIG_FIRST_CAM).equals(uid)) {
+            addCamera(uid);
+        } else if (config.get(CONFIG_SECOND_CAM).equals(uid)) {
+            addCamera(uid);
+        } else if (config.get(CONFIG_THIRD_CAM).equals(uid)) {
+            addCamera(uid);
+        } else if (config.get(CONFIG_FORTH_CAM).equals(uid)) {
+            addCamera(uid);
+        }
+    }
+
+    // Event based. This is called as each camera comes online after the group handler is registered.
+    public void cameraOffline(IpCameraHandler handle) {
+        if (cameraOrder.remove(handle)) {
+            logger.info("Camera {} is now offline, now removed from this group.", handle.getThing().getUID().getId());
+        }
+    }
+
+    boolean addIfOnline(String UniqueID) {
+        if (IpCameraHandler.listOfOnlineCameraUID.contains(UniqueID)) {
+            addCamera(UniqueID);
+            return true;
+        }
+        return false;
+    }
+
+    void createCameraOrder() {
+        addIfOnline(config.get(CONFIG_FIRST_CAM).toString());
+        addIfOnline(config.get(CONFIG_SECOND_CAM).toString());
+        if (config.get(CONFIG_THIRD_CAM) != null) {
+            addIfOnline(config.get(CONFIG_THIRD_CAM).toString());
+        }
+        if (config.get(CONFIG_FORTH_CAM) != null) {
+            addIfOnline(config.get(CONFIG_FORTH_CAM).toString());
+        }
+        // Cameras can now send events of when they go on and offline.
+        IpCameraHandler.listOfGroupHandlers.add(this);
+    }
+
+    int checkForMotion(int nextCamerasIndex) {
+        int checked = 0;
+        for (int index = nextCamerasIndex; checked < cameraOrder.size(); checked++) {
+            if (cameraOrder.get(index).motionDetected) {
+                // logger.trace("Motion detected on a camera in a group, the display order has changed.");
+                return index;
+            }
+            if (++index >= cameraOrder.size()) {
+                index = 0;
+            }
+        }
+        return nextCamerasIndex;
+    }
+
+    Runnable pollingCameraGroup = new Runnable() {
+        @Override
+        public void run() {
+            if (cameraOrder.isEmpty()) {
+                createCameraOrder();
+            }
+            if (++cameraIndex >= cameraOrder.size()) {
+                cameraIndex = 0;
+            }
+            if (motionChangesOrder) {
+                cameraIndex = checkForMotion(cameraIndex);
+            }
+            if (hlsTurnedOn) {
+                discontinuitySequence++;
+                createPlayList();
+                if (mediaSequence > 2147000000) {
```

Comment:
Removed it as I just learnt Java gracefully handles overflows. Used to program microprocessors.

Reply:
Java handles overflows like microcontrollers. They simply overflow.

Reply:
I'm confused now and need someone that knows Java to tell me what to do... Yes they both overflow but at least in the old days when a variable/register overflowed it would effect the next memory position and cause undesirable effects. So really all I need to know is, do I have to check and prevent the overflow, or is it safe to let an overflow occur in Java?

Reply:
Maybe there's a misunderstanding, but an integer overflow doesn't affect the next memory position, at least in architectures like Java, x86, AVR, MSP430, ARM M0+. It overflows to 0 after the max value is reached. So, you don't need to handle overflows to protect any memory. But it might be necessary from the functional perspective as the value drops to 0 suddenly when overflowing, and starts over.

Reply:
Thank you, the code will work fine unless someone wants to stream non stop for more than 136 years :)